server/container/guice/guice-common/src/main/java/org/apache/james/GuiceJamesServer.java

```diff
@@ -88,12 +75,11 @@ public void stop() {
         }
     }
 
-    public ExtendedServerProbe serverProbe() {
-        return serverProbe;
+    public GuiceServerProbe serverProbe() {
+        return guiceProbeProvider.getProbe(GuiceServerProbe.class);
     }
 
-    public Optional<Port> getWebadminPort() {
-        return webadminPort;
+    public GuiceProbeProvider getGuiceProbeProvider() {
```

Comment:
Well, I really like the idea the server not to have any idea of the probes it have.

Anyway, we can make this protected.

We then introduce two interfaces : 
 -  JmapServer (defines a getter on JmapGuiceProbe)
 -  WebAdminServer (defines a getter on WebAdmin probe)

We can  then have a CassandraGuiceJamesServer + MemoryGuiceJamesServer implementing these interfaces.

Abstract integration tests for Jmap can then ask for a GuiceJamesServer implementing JmapServer (idem for webadmin).

Would that seem Ok to you ?